### PR TITLE
add openid token

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1120

### Change description ###

add getting openid token

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -135,12 +135,12 @@ bintray {
...
-    feignVersion = '10.2.3'
-    feignFormVersion = '3.5.0'
-    wiremockVersion = '2.24.1'
+    feignVersion = '10.9'
+    feignFormVersion = '3.8.0'
+    wiremockVersion = '2.26.3'
 }
 
-ext["spring-cloud-openfeign.version"] = "2.1.2.RELEASE"
+ext["spring-cloud-openfeign.version"] = "2.2.2.RELEASE"
```
Comment (doncem) [inline build.gradle:143]: how about reviewing existing PR first? #65

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/TokenRequest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    public String getRefreshToken() {
+        return refreshToken;
+    }
+
+    public String getCode() {
+        return code;
+    }
+}
+
+
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/idam/client/models/TokenRequest.java:86]: I think project uses lombok api to minify such declarations as POJO, for example

File: src/test/java/uk/gov/hmcts/reform/idam/client/IdamClientTest.java [hunk 3]
```diff
@@ -54,8 +58,27 @@
...
+        + "ZXJuYWw6ODQ0My9vcGVuYW0vb2F1dGgyL2htY3RzIiwidG9rZW5OYW1lIjoicmVmcmVzaF90b2tlbiIsInRva2Vu3R5cGUiOiJCZW"
+        + "EpVQst7LMVefQuZTkSD3KQ_YBnnKuicZc4mD-QLLOCY-InfC_P7Sf6ZloNiEX4H-WquApx9HLTHPNRsH3TNvJgWEJbRlpkE5dWcBe"
+        + "fyJdXaI5YUO6bNxmby3jkqRVUMe25nOwWMiOyqoZDg3ehMGJuSqMguQwMrg1kc8RB6ZtVugPctVgW_ffE9EYc3i8yfTqq8rUvxDJI"
+        +  "dCT4jkArwuitcvSmUg2XCTy_YoqdwQcGZD5vI3Wya1polA";
+    private final String ID_TOKEN = "eyJ0eXAiOiJKV1QiLCJraWQiOiIxZXIwV1J3Z0lPVEFGb2pFNHJDL2ZiZUt1M0k9IiwiYWxnIj"
+        + "oiUlMyNTYifQ.eyJhdF9oYXNoIjoibFpJWlg1M3BzRnVCZWlpMllLWTFBUSIsInN1YiI6ImRpdm9yY2UtYnNwLXN5c3RlbS11cGR"
+        + "hdGVAaG1jdHMubmV0IiwiYXVkaXRUcmFja2luZ0lkIjoiZjdkMDhmZjAtNDI1Ny00MjczLTkxOGMtMGNmOTJkZGEyN2NiEwNDEzN"
+        + "XQiOjE1ODY5NTI5NDB9.ZpCCIHT98QbKjNeJzALhwL2yjnHBbC4PLbGrrQaVi3thcZ5_TaGmSxcECTgpjGTESJbccvlDe7z1xihO"
+        + "Vmxt1h2dGD9dAPBYSR6G0LEP_N5MUUCahVMDQeSBawzwW54AOsm4wwd5UUV_Xn8tAvovt4g-iZQGwBsi6t_FTLLYiPzapL-12jKt"
+        + "oCKtFmyLlfcBXTLaPywi8oFfinCRaVQ83BiKtIXuImGrYN8WeZVtvZwAQzmHqA4PoDJBzOJWptL-Z63wVFoQZy2AHaFLcR_Yv07w";
```
Comment (doncem) [inline src/test/java/uk/gov/hmcts/reform/idam/client/IdamClientTest.java:74]: i hope these don't hold any sensitive information. i tried to decode but couldn't get anything :/

File: build.gradle (hunk 1 above)
Comment (mdayican) [reply build.gradle:143]: I need to use new form version, as consequence of it I changed them

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/TokenRequest.java (hunk 2 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/idam/client/models/TokenRequest.java:86]: did not work with formproperty

File: src/test/java/uk/gov/hmcts/reform/idam/client/IdamClientTest.java (hunk 3 above)
Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/idam/client/IdamClientTest.java:74]: it is kind of made up

**Comment by mdayican** [comment]
> 8 spaces instead of 4 are all over this pr

reformatted

File: build.gradle (hunk 1 above)
Comment (doncem) [reply build.gradle:143]: aye, just noticed there's a new annotation. fixing #65 as well. with feign 10.9 form also comes in as 3.8 - there is no need to include it separately in dependency set. check out that pr - should be all clear and help out this one as well

File: src/main/java/uk/gov/hmcts/reform/idam/client/models/TokenRequest.java (hunk 2 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/idam/client/models/TokenRequest.java:86]: tried locally - seems to be working. unless i need to verify it somehow differently

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/idam/client/models/TokenRequest.java:53]: added

**Pull request merged**